R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -1964,6 +1964,38 @@ test_that("Method str()", {
   expect_equal(capture.output(utils:::str(iris)), capture.output(str(iris)))
 })
 
+test_that("dapply() on a DataFrame", {
+  df <- createDataFrame (sqlContext, mtcars)
+  df1 <- dapply(df, function(x) { x }, schema(df))
+  result <- collect(df1)
+  expected <- mtcars
+  rownames(expected) <- NULL
+  expect_identical(expected, result)
+
+  df1 <- dapply(df, function(x) { x + 1 }, schema(df))
+  result <- collect(df1)
+  expected <- mtcars + 1
+  rownames(expected) <- NULL
+  expect_identical(expected, result)
+
+  schema <- structType(structField("a", "double"), structField("b", "double"))
+  df1 <- dapply(df, function(x) { x[, 1:2, drop = F] }, schema)
+  result <- collect(df1)
+  expected <- mtcars[, 1:2, drop = F]
+  names(expected) <- c("a", "b")
+  rownames(expected) <- NULL
+  expect_identical(expected, result)
+
+  df1 <- dapply(df, function(x) { x + 1 })
+  schema <- structType(structField("a", "double"), structField("b", "double"))
+  df2 <- dapply(df1, function(x) { x[, 1:2, drop = F] }, schema)
+  result <- collect(df2)
+  expected <- (mtcars + 1)[, 1:2, drop = F]
+  names(expected) <- c("a", "b")
+  rownames(expected) <- NULL
+  expect_identical(expected, result)
+})
```

Comment:
Also, I think that it would be good to add different data types beside double in the test case.